src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -290,24 +275,121 @@ private async Task ProcessQueueAsync()
     }
 
     /// <summary>
-    /// Allows XAML to inspect the request before its dispatched.
-    /// Should not generally be used, this will be replaced by the OOP XAML server.
+    /// Reflection invokes <see cref="ProcessQueueCoreAsync{TRequest, TResponse}(IQueueItem{TRequestContext}, IMethodHandler, RequestHandlerMetadata, ConcurrentDictionary{Task, CancellationTokenSource}, CancellationTokenSource?, CancellationToken)"/>
+    /// using the concrete types defined by the handler's metadata.
     /// </summary>
-    [Obsolete("Only for use by legacy XAML LSP")]
-    protected internal virtual void BeforeRequest<TRequest>(TRequest request)
+    private async Task InvokeProcessCoreAsync(
+        IQueueItem<TRequestContext> work,
+        RequestHandlerMetadata metadata,
+        IMethodHandler handler,
+        MethodInfo methodInfo,
+        ConcurrentDictionary<Task, CancellationTokenSource> concurrentlyExecutingTasks,
+        CancellationTokenSource? currentWorkCts,
+        CancellationToken cancellationToken)
     {
-        return;
+        var task = methodInfo.Invoke(this, [work, handler, metadata, concurrentlyExecutingTasks, currentWorkCts, cancellationToken]) as Task
+            ?? throw new InvalidOperationException($"ProcessQueueCoreAsync result task cannot be null");
+        await task.ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Given a concrete handler and types, this dispatches the current work item to the appropriate handler,
+    /// waiting or not waiting on results as defined by the handler.
+    /// </summary>
+    private async Task ProcessQueueCoreAsync<TRequest, TResponse>(
+        IQueueItem<TRequestContext> work,
+        IMethodHandler handler,
+        RequestHandlerMetadata metadata,
+        ConcurrentDictionary<Task, CancellationTokenSource> concurrentlyExecutingTasks,
+        CancellationTokenSource? currentWorkCts,
+        CancellationToken cancellationToken)
+    {
+        var deserializedRequest = work.TryDeserializeRequest<TRequest>(_languageServer, metadata, handler.MutatesSolutionState);
```

Comment:
comment that if htis fails, we log it already.  if we're not logging this, we shou.d